### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by isidzukuri**

### CODE REVIEW

**Comment by isidzukuri** [comment]
users and admins authorization

File: Gemfile [hunk 1]
```diff
@@ -46,3 +46,4 @@ end
 
 # Windows does not include zoneinfo files, so bundle the tzinfo-data gem
 gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
+gem 'devise'
```
Comment (BLauris) [inline Gemfile]: It's a good idea to add gem version.

File: app/controllers/categories_controller.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class CategoriesController < ApplicationController
+  protect_from_forgery with: :exception
+
+  def index
```
Comment (BLauris) [inline app/controllers/categories_controller.rb]: Why here is an empty action?

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -9,6 +9,15 @@
   </head>
```
Comment (BLauris) [inline app/views/layouts/application.html.erb]: please ass gem "haml" and write in hamn

File: app/controllers/categories_controller.rb (hunk 2 above)
Comment (isidzukuri) [inline app/controllers/categories_controller.rb]: fixed

File: Gemfile (hunk 1 above)
Comment (isidzukuri) [inline Gemfile]: fixed

File: app/views/layouts/application.html.erb (hunk 3 above)
Comment (isidzukuri) [inline app/views/layouts/application.html.erb]: done

File: app/views/layouts/application.html.haml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+!!!
+%html
+  %head
+    %title ECrawler
+    = csrf_meta_tags
+    = stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload'
+    = javascript_include_tag 'application', 'data-turbolinks-track': 'reload'
+  %body
+    %ul.right
+      - if signed_in?
```
Comment (AKovtunov) [inline app/views/layouts/application.html.haml]: better to create a partial for header and\or user actions

File: test/models/role_test.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class RoleTest < ActiveSupport::TestCase
```
Comment (AKovtunov) [inline test/models/role_test.rb]: empty tests. write them or delete files

File: test/models/user_test.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+  # test "the truth" do
```
Comment (AKovtunov) [inline test/models/user_test.rb]: empty tests. write them or delete files

File: app/views/layouts/application.html.haml (hunk 4 above)
Comment (isidzukuri) [inline app/views/layouts/application.html.haml]: sure

Comment (isidzukuri) [inline app/views/layouts/application.html.haml]: done

File: test/models/role_test.rb (hunk 5 above)
Comment (isidzukuri) [inline test/models/role_test.rb]: fixed

File: test/models/user_test.rb (hunk 6 above)
Comment (isidzukuri) [inline test/models/user_test.rb]: fixed

**Pull request merged**